### PR TITLE
quantum-expresso: fix for Fujitsu compiler

### PR DESCRIPTION
**Pull request by tkameyama**
- configure update to detect Fujitsu fortran compiler.
- fix fox configure.
fox's configure search -out and -o option to specify Fortran output file.
Fujitsu fortran accept -out option, but the output file cannot execute.
- fix scalapack detection.
When configue --scalapack=yes is specifies, configure append only -lscalapack option to linker.
But fujitsu SSL2 library on a64fx must use -lscalapacksve. So this PR set SCALAPACK_LIBS option.
- fix intent and AND function usage.

### CODE REVIEW

**Review by tldahlgren — APPROVED** [review]
LGTM.  

Approving under assumption approved by the package maintainer.

File: var/spack/repos/builtin/packages/quantum-espresso/package.py [hunk 1]
```diff
@@ -244,9 +254,28 @@ class QuantumEspresso(Package):
...
     # Configure updated to work with NVIDIA compilers
     patch('nvhpc.patch', when='@6.5 %nvhpc')
 
+    # Configure updated to work with Fujitsu compilers
+    patch('fj.6.5.patch', when='@6.5 %fj')
+    patch('fj.6.6.patch', when='@6.6:6.7 %fj')
+
+    # extlibs_makefile updated to work with fujitsu compilers
+    patch('fj-fox.patch', when='%fj')
+
```
Comment (naromero77) [inline var/spack/repos/builtin/packages/quantum-espresso/package.py]: The `when=` clauses here need a `+patch`

**Comment by naromero77** [comment]
Also, would be good to get these patches upstreamed.

File: var/spack/repos/builtin/packages/quantum-espresso/package.py (hunk 1 above)
Comment (naromero77) [reply var/spack/repos/builtin/packages/quantum-espresso/package.py]: I should add some context here. There are some 3rd party patches in QE. The one I added is for QMCPACK. We should really just upstream this code but it never happened for various reasons. The `+patch` which is the default variant is just a way to detect/label that there is a patch on the official release version of QE. So if you apply a patch from upstream or elsewhere, you should include `when='+patch'` to activate the patch.

So, `when='~patch'`, means plain vanilla QE with no patches.

Its not a great system at all, but there is no way to suppress patch on a per patch level in Spack.

Comment (tkameyama) [reply var/spack/repos/builtin/packages/quantum-espresso/package.py]: fixed.

**Review by naromero77 — APPROVED** [review]

**Comment by tldahlgren** [comment]
Thank you for your contribution.

**Pull request merged**